### ISSUE TITLE
Indexing: (Re)index selected photos/videos

### ISSUE THREAD

**Issue by dgtal1** [photoprism/photoprism#4186]
**Describe what problem this solves and why this would be valuable to many users**

For some reason I've got some issues in my index and some photos do not have their location extracted or some videos not visible in the Places view despite having the location data. I've uploaded these originals I noticed were missing some data or missing in the search view again and then run reindexing (but without a complete rescan as it takes ages to complete), but it didn't solve it.

**Describe the solution you'd like**

As indexing issues may occur at any time for many different reasons - I think it would make sense if it was possible to be able to force the indexing process, but only limited to selected files. Even the simplest solution, which would be adding CLI support would be fine for me. Obviously the best solution would be to select the files in the UI and then run some action for them by pushing a button.

Indexing is a super-resource heavy process and takes ages on slower hardware like NAS devices or RPis, thus reducing the indexing time by only executing it for limited dataset would save a lot of time and energy.

**Describe alternatives you've considered**

Full, forced reindex is not an option as it's taking many days to be run on my NAS. It takes 3 days on my PC when running 24 hours a day. No other way of fixing the index issues that I'm aware of.

**Additional context**

Screenshot showing a file, that is not shown in the Places view, despite having the location data. The reindex could be an action covered by the contextual button which allows metadata edits etc:
![image](https://github.com/photoprism/photoprism/assets/27864579/a8523963-1400-4b08-930f-e70bbe11f711)

The file has location data, but Photoprism missed it somewhere and it wasn't indexed properly. I did not add this manually:
![image](https://github.com/photoprism/photoprism/assets/27864579/3448071d-7588-4354-9fff-765d2230c6aa)

I expect the solution to cover this case - basically to process te file and extract & process whatever is extracted and processed during the full, forced reindexing process (location, photo name, tagging/labels etc).

PS. I've just noticed the video file does not have the country name despite having the location data - apparently wasn't extracted / processed. I guess something went wrong in the processing chain and the data were extracted, but not processed.

**Comment by lastzero** [comment]
While you cannot reindex single files, you can search and select folders for indexing as shown in this screenshot taken on demo.photoprism.app:
![IMG_4860](https://github.com/photoprism/photoprism/assets/301686/8d2f3443-251d-4bc5-80a6-9ec5303cd3e7)

Unless you happen to have all your files in a single directory (which may also cause filesystem related performance issues), that should provide you with an easy way to only reindex certain files. Please note that a complete rescan is generally not required and only recommended in case of problems or a major update.

**Comment by rob-miller** [comment]
I'm not sure if this should be a new issue, but I start here as it seems most closely related.

My directory structure (coming from phockup) is year/month/day, because I have some days with a few thousand photos.  My photo library goes back to 2002.

It is painful to scroll down the menu shown above to get to 2024, both because of the number of list entries and the list is slow to populate.  In Firefox, I can type in the textbox but when I click outside the box it reverts to whatever was there before.  The only way I can get '2024' in the box at the top is using the scrollwheel, the scrollbar doesn't get past what has been loaded by spinning my mousewheel.

Sorry if I missed this in some other issue.

**Comment by lastzero** [comment]
Any reason you don't click inside the box?

**Comment by rob-miller** [comment]
Yes, I click inside the box and attempt to type as described above. When I
click outside the box, the text reverts to what was last in the box.

**Comment by lastzero** [comment]
Does it work with Chrome?

**Comment by rob-miller** [comment]
The behaviour is the same on Chrome, Safari (I'm on a Mac), and Firefox, but I have figured it out.

You type inside the box, then click on the entry in the drop-down - even if it is the same as what you have already typed in the box.  Here are some examples that I see (in Chrome):

Start from "All originals"
type /2024  (optionally followed by **enter** or **tab)**
/2024 appears in dropdown
click outside dropdown -> back to All Originals

Start from "All originals"
type /2024
click on /2024 in dropdown
dropdown closes -> /2024 selected

Less important:
Start from "All originals"
type /2022
scroll dropdown to /2022/12/29
type /2023
dropdown is at /2023/12/29 (updated year but not at beginning of year)
type /2023/01
dropdown is at 2023/01/29
then more back/forward typing I can't quite quantify will get dropdown to show /2023 only so can select.

While I understand that clicking outside the dropdown reasonably means "no I didn't want that",  other arguments are 
- pressing **enter** at the end means I really do want just what I typed (if it is valid)
- I typed what I want, it's validated in the dropdown, don't make me use the mouse to select the same thing again (actually just need down arrow to avoid mouse)

But I see that it is just UI preferences and surprises/expectations, sorry for the noise.
